src/coreclr/tools/aot/crossgen2/Program.cs

```diff
@@ -253,6 +253,22 @@ public int Run()
                 }
             }
 
+            Dictionary<string, AssemblyNameInfo> allAssemblyNames = new ();
```

Comment:
I think this is too late to detect duplicates.

crossgen, ilc and ilverify are all on first reference wins plan for references and ignore duplicates.

The duplicates are eliminated very early during command line parsing at https://github.com/dotnet/runtime/blob/0621e649bd084cb0dfd1f2e627538e7d9aa9e211/src/coreclr/tools/Common/CommandLineHelpers.cs#L347-L358.

`strict` flag is true for input assemblies and it is false for references.

Reply:
Thanks for investigating this. I'll figure out whether it's worth doing the work to actually detect this scenario robustly and generate warnings for it.